### DISCUSSION TITLE
Question about "eval_every_steps"

### DISCUSSION THREAD

**Discussion by eleftheriad** [nyu-mll/jiant#1333] [Q&A] [answered]
Hi, quick question

Does "eval_every_steps" dictate when the model updates its weights or is it only for the early stopping?

**Comment by zphang** [comment, marked as answer]
`eval_every_steps` is for both early stopping as well as recording of metrics. It should be unrelated to when the model updates its weights.

**Reply by eleftheriad** [reply]
So the weights are still being updated every one epoch, whatever the value on "eval_every_steps" may be?

**Reply by zphang** [reply]
The weights are updated every batch, and should be independent of `eval_every_steps`.
